minion/kubernetes/configmap.go

```
@@ -0,0 +1,122 @@
+package kubernetes
+
+import (
+	"crypto/sha1"
+	"fmt"
+
+	"github.com/kelda/kelda/blueprint"
+	"github.com/kelda/kelda/db"
+	"github.com/kelda/kelda/join"
+	"github.com/kelda/kelda/util/str"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// updateConfigMaps creates a config map for each container's
+// FilepathToContent. It ignores any Secret ContainerValues.
+// These config maps are referenced by the updateDeployment function to mount
+// the container's filepathToContent into the container.
+func updateConfigMaps(conn db.Conn, configMapsClient clientv1.ConfigMapInterface) (
+	noErrors bool) {
```

This seems odd, why not just return an error?

---

It's to avoid the complexity of aggregating multiple errors. The caller wouldn't make use of the error anyways, it would just check whether it was `== nil`. What do you think?